lib/TPP/Dialect/Tpp/TppUtils.cpp

```diff
@@ -343,22 +345,41 @@ static bool isUnaryOp(linalg::GenericOp linalgOp) {
   return false;
 }
 
-static bool hasMaxfZeroOp(linalg::LinalgOp linalgOp) {
+static bool matchReluBody(linalg::LinalgOp linalgOp, OperandInfo &info) {
   if (!isa<linalg::GenericOp>(linalgOp))
     return false;
 
   auto genOp = cast<linalg::GenericOp>(linalgOp);
   if (!hasOnlyOp<arith::MaxFOp>(genOp.getRegion()))
     return false;
 
-  for (Operation &op : genOp.getRegion().front()) {
-    if (auto maxfOp = dyn_cast_or_null<arith::MaxFOp>(op)) {
-      // Check both operands if either one is a zero filled tensor.
-      if (isZeroTensor(maxfOp.getLhs()) || isZeroTensor(maxfOp.getRhs()))
-        return true;
-    }
-  }
+  Operation *regionOp = &genOp.getRegion().front().front();
+  auto maxfOp = cast<arith::MaxFOp>(regionOp);
+  Value maxfLhs = maxfOp.getLhs();
+  Value maxfRhs = maxfOp.getRhs();
+
+  if (isZeroTensor(maxfLhs) && isZeroTensor(maxfRhs))
```

Comment:
This looks like a silly special case, and to be honest, `maxf(0.0, 0.0)` is still "technically" a ReLU.

In any case, lowering that to TPP would be faster than defaulting to loops, so I think it's best to just let this one slide into the cases below.

Reply:
Perhaps mathematically it counts as a ReLU if you squint hard enough. But in reality it is just a zero fill of the output buffer so, I imagine a simple memset should be faster than TPP.

Reply:
*than `tpp.relu`. Although, okay - I admit I'm not 100% sure about this claim. And there is a TPP Zero operation which could be better.
Anyway, if we apply `tpp.relu` on a random uninitialized buffer, we might end up with a different result than zero initialization.

Reply:
> on a random uninitialized buffer, we might end up with a different result than zero initialization

"undefined behaviour is undefined" :smile: 

> And there is a TPP Zero operation which could be better.

Another reason why mapping one by one, running all comparators for all options won't scale.